x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DelayedDataDetector.java

```diff
@@ -38,24 +41,23 @@
 public class DelayedDataDetector {
 
     private static final String DATE_BUCKETS = "date_buckets";
+    private static final int FALLBACK_NUMBER_OF_BUCKETS_TO_SPAN = 7;
     private final long bucketSpan;
     private final long window;
     private final DatafeedConfig datafeedConfig;
     private final Client client;
     private final Job job;
 
-    public DelayedDataDetector(Job job, DatafeedConfig datafeedConfig, TimeValue window, Client client) {
+    public DelayedDataDetector(Job job, DatafeedConfig datafeedConfig, Client client) {
         this.job = job;
         this.bucketSpan = job.getAnalysisConfig().getBucketSpan().millis();
         this.datafeedConfig = datafeedConfig;
-        long windowMillis = window.millis();
-        if (windowMillis < bucketSpan) {
-            throw new IllegalArgumentException("[window] must be greater or equal to the [bucket_span]");
-        }
-        if (Intervals.alignToFloor(windowMillis/bucketSpan, bucketSpan) >= 10000) {
-            throw new IllegalArgumentException("[window] must contain less than 10000 buckets at the current [bucket_span]");
+
+        if (datafeedConfig.getShouldRunDelayedDataCheck()) {
```

Comment:
What if we didn't check this in the detector at all and we simply check this in the `DatafeedJob.checkForMissingDataIfNecessary()`? I think it would make this class cleaner. Alternatively, we could also have a `NullDelayedDataDetector` implementation that always returns an empty list when check is invoked. The latter would allow us to sanity check the `window` value too.